crates/networking/p2p/discv4/server.rs

```diff
@@ -384,6 +362,33 @@ impl DiscoveryServer {
         Ok(())
     }
 
+    async fn send_enr_request(&mut self, node: &Node) -> Result<(), DiscoveryServerError> {
+        let expiration: u64 = get_msg_expiration_from_seconds(EXPIRATION_SECONDS);
+        let enr_request = Message::ENRRequest(ENRRequestMessage { expiration });
+
+        let mut buf = Vec::new();
+        enr_request.encode_with_header(&mut buf, &self.signer);
+        let enr_request_hash: [u8; 32] = buf[..32]
+            .try_into()
+            .expect("first 32 bytes are the message hash");
```

Comment:
It would probably be more convenient to just return the hash from `encode_with_header`.